### PR TITLE
Initial merge spec

### PR DESCRIPTION
**Pull request by timbeiko**
Adds an initial spec for the merge, given that we now have 2 EIPs, 3 devnets and a spec for the Engine API. 

#### Cute Animal Picture
My dog wants me to shut down my computer 👋 
IMG_0374

### CODE REVIEW

File: network-upgrades/mainnet-upgrades/merge.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## Merge Upgrade Specification
```
Comment (lightclient) [inline network-upgrades/mainnet-upgrades/merge.md:1]:
```suggestion
## Serenity Upgrade Specification
```
I think you misspelled Serenity.

Comment (timbeiko) [reply network-upgrades/mainnet-upgrades/merge.md:1]: tfw you try and go for the name-agnostic spec...!

Comment (lightclient) [reply network-upgrades/mainnet-upgrades/merge.md:1]: oof saw this before i saw your discord post haha

Comment (djrtwo) [reply network-upgrades/mainnet-upgrades/merge.md:1]: Serenity is (if anything) the meta-name!

Serenity is the effort to make ethereum more secure, sustainable, and scalable, but the Merge only handles the first two

File: network-upgrades/mainnet-upgrades/merge.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+### Included EIPs
+Specifies changes included in the Network Upgrade.
+
+  - [x] [EIP-3675: Upgrade consensus to Proof-of-Stake](https://eips.ethereum.org/EIPS/eip-3675)
+  - [x] [EIP-4399: Supplant DIFFICULTY opcode with RANDOM](https://eips.ethereum.org/EIPS/eip-4399)
+
+### Engine API
+
+A new set of APIs is introduced as part of The Merge for the execution layer clients to communicate with the consensus layer. The specification for it is available [here](https://github.com/ethereum/execution-apis/tree/main/src/engine).
+
```
Comment (djrtwo) [inline network-upgrades/mainnet-upgrades/merge.md:12]: We might include CL specs as a section.

*If* an EL upgrade requires an accompanying CL upgrade, then I think it makes sense for the specs to cross-reference eachother

Comment (timbeiko) [reply network-upgrades/mainnet-upgrades/merge.md:12]: Added a link, good call!

**Pull request merged**